### PR TITLE
Residents Directory - Delete Pagination Render

### PR DESCRIPTION
**Pull request by achandrabalan**
## Implementation description
* Deleting resident automatically removes log from modal, not after refresh
* Deleting resident appropriately updates user page number, resident count, get records

## Steps to test
Deleting a resident..

1. Triggers toast message
2. Decrements resident count by 1
3. If it's the last resident on the page, the user is brought to the previous page. Otherwise, all resident below deleted one are shifted up

## What should reviewers focus on?
* 

## Checklist
- [ ] My PR name is descriptive and in imperative tense
- [ ] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [ ] I have run the appropriate linter(s)
- [ ] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR
- [ ] If I have made API changes, I have updated the [REST API Docs](https://www.notion.so/uwblueprintexecs/REST-Endpoints-05ce60312bb943439dfda42bb1318536)
- [ ] IF I have made changes to the db/models, I have updated the [Data Models Page](https://www.notion.so/uwblueprintexecs/Data-Models-760f8aa06b244eb0842c079ad77987b0)
- [ ] I have documented this PR in the [Production Release Page](https://www.notion.so/uwblueprintexecs/fe28a97bfa5648d3bc3795b32b694acd?v=5565cfb35dcc45bb84f6528cb09517cd)
- [ ] I have updated other Docs as needed

### CODE REVIEW

**Review by owen-sellner — APPROVED** [review]
LGTM!!!

File: frontend/src/components/forms/CreateResident.tsx [hunk 1]
```diff
@@ -30,14 +30,26 @@ import { singleDatePickerStyle } from "../../theme/forms/datePickerStyles";
...
 // TODO: Connect to Buidings table
 const BUILDINGS = [
   { label: "144", value: "144" },
   { label: "362", value: "362" },
   { label: "402", value: "402" },
 ];
 
-const CreateResident = (): React.ReactElement => {
+const CreateResident = ({
+  userPageNum,
```
Comment (connor-bechthold) [inline frontend/src/components/forms/CreateResident.tsx]: I don't believe you need to pass in `userPageNum` here

File: frontend/src/components/pages/ResidentDirectory/ResidentDirectoryTable.tsx [hunk 2]
```diff
@@ -99,6 +107,14 @@ const ResidentDirectoryTable = ({
         "Resident has been deleted successfully.",
         "success",
       );
+      const newUserPageNum: number = (
```
Comment (connor-bechthold) [inline frontend/src/components/pages/ResidentDirectory/ResidentDirectoryTable.tsx]: `number` can be removed here

**Review by connor-bechthold — CHANGES_REQUESTED** [review]
beauty saw a few small things to modify but you should be good to go after 👌

File: frontend/src/components/pages/ResidentDirectory/ResidentDirectoryTable.tsx [hunk 3]
```diff
@@ -174,7 +190,11 @@ const ResidentDirectoryTable = ({
           <EditResident
             resident={editingResident}
             isOpen={isEditModalOpen}
-            toggleClose={handleEditClose}
+            toggleClose={() => {
+              handleEditClose()
+              getRecords(userPageNum)
```
Comment (connor-bechthold) [inline frontend/src/components/pages/ResidentDirectory/ResidentDirectoryTable.tsx:195]: Should also countRecords here in case new records have been added

Comment (connor-bechthold) [inline frontend/src/components/pages/ResidentDirectory/ResidentDirectoryTable.tsx]: You'd want to pass this as a prop to the `EditResident` component and then call it right after the resident is successsfully updated

File: frontend/src/components/forms/CreateResident.tsx [hunk 4]
```diff
@@ -132,7 +146,7 @@ const CreateResident = (): React.ReactElement => {
     }
 
     addResident();
-
+    setUserPageNum(1);
```
Comment (connor-bechthold) [inline frontend/src/components/forms/CreateResident.tsx]: I would just add this to `getResidents` where you called get and count above on line 74-75 to keep things consistent

**Review by connor-bechthold — APPROVED** [review]
LGTM 💯

**Pull request merged**